src/tarantool-gdb.py

```diff
@@ -1399,3 +1415,540 @@ def invoke(self, arg, from_tty):
+        entry_type, _, fields = str.partition('::')
+        self.__entry_type = gdb.lookup_type('struct ' + entry_type)
+        self.__offset = 0
+        container_type = self.__entry_type
+        for field in fields.split('::'):
+            self.__offset -= int(container_type[field].bitpos/8)
+            container_type = container_type[field].type
+
+    def __eq__(self, other):
+        return self.__str == other.__str
+
+    def __hash__(self):
+        return hash(str(self))
+
+    def __str__(self):
+        return self.__str
+
+    @property
+    def entry_type(self):
+        return self.__entry_type
+
+    def entry_from_item(self, item):
+        assert equal_types(item.type.target(), TtList.rlist_type)
+        return cast_ptr(self.__entry_type, item, self.__offset)
+
+class TtList:
+    rlist_type = gdb.lookup_type('rlist')
+
+    def __init__(self, head, entry_info=None):
+        assert equal_types(head.type.target(), self.rlist_type)
+        self.__head = head
+        self.__entry_info = entry_info
+        self.__len = -1
+
+    @property
+    def entry_info(self):
+        return self.__entry_info
+
+    def __iter__(self):
+        item = self.__head['next']
+        while item != self.__head:
+            yield item
+            item = item['next']
+
+    def __reversed__(self):
+        item = self.__head['prev']
+        while item != self.__head:
+            yield item
+            item = item['prev']
+
+    def __len__(self):
+        if self.__len == -1:
+            self.__len = 0
+            for _ in self:
+                self.__len += 1
+        return self.__len
+
+    def entry_from_item(self, item):
+        assert equal_types(item.type.target(), self.rlist_type)
+        return self.__entry_info.entry_from_item(item)
+
+class TtListsLut:
+    def build_list_variables_map():
+        ret = {}
+        for sym, entry in [
+            ('box_on_call', 'trigger::link'),
+            ('box_on_select', 'trigger::link'),
+            ('box_on_shutdown_trigger_list', 'trigger::link'),
+            ('box_raft_on_broadcast', 'trigger::link'),
+            ('engines', 'engine::link'),
+            ('log_rotate_list', 'log::in_log_list'), #static
+            ('on_access_denied', 'trigger::link'),
+            ('on_alter_func', 'trigger::link'),
+            ('on_alter_sequence', 'trigger::link'),
+            ('on_alter_space', 'trigger::link'),
+            ('on_console_eval', 'trigger::link'),
+            ('on_schema_init', 'trigger::link'),
+            ('on_shutdown_trigger_list', 'on_shutdown_trigger::link'),
+            ('popen_head', 'popen_handle::list'),
+            ('replicaset_on_quorum_gain', 'trigger::link'),
+            ('replicaset_on_quorum_loss', 'trigger::link'),
+            ('session_on_auth', 'trigger::link'),
+            ('session_on_connect', 'trigger::link'),
+            ('session_on_disconnect', 'trigger::link'),
+            ('shutdown_list', 'session::in_shutdown_list'),
```

Comment:
I'd like to remind that initially it was constant and I put it inline to tiе tuple composition and decomposition as close as possible instead of introducing named tuple. For that reason I would keep it inline.